packages/labs/motion/src/animate.ts

```diff
@@ -192,6 +192,9 @@ export class Animate extends AsyncDirective {
     const firstUpdate = this._host === undefined;
     if (firstUpdate) {
       this._host = part.options?.host as LitElement;
+      if (this._host == null) {
+        throw new Error("animate directive must be used in a LitElement");
```

Comment:
The host just has to implement the ReactiveControllerHost interface. So it can be a ReactiveElement or other implementation.

Reply:
Thanks for checking @justinfagnani! Yes, that's what I thought at the beginning. I've implemented [HookComponents](https://fable.io/Fable.Lit/docs/hook-components.html) (similar to VirtualComponent in Haunted) using `AsyncDirective` in lit 2, and tried to implement the [ReactiveControllerHost interface](https://lit.dev/docs/api/controllers/#ReactiveControllerHost). But this was not enough for `animate` because it depends on methods that are not on the interface but come from ReactiveElement class, like `hasUpdated` or `updateCompleted`: https://github.com/lit/lit/blob/fba16cc1ceac30184514d32cd8402f9086e0eb43/packages/labs/motion/src/animate.ts#L445-L453

Actually I wanted to ask about this, because if Reactive Controllers stick to the ReactiveControllerHost interface they could be used more broadly (e.g. you could create a React hook that accepts reactive controllers), but if controllers are going to depend on LitElement at the end this won't be the case (and it makes the `ReactiveControllerHost` interface somewhat useless).